### PR TITLE
Fix control-plane generation for headers containing structs

### LINKED ISSUE #3329: P4Runtime generation does not support generic structs
I was testing the compiler with `--p4runtime-files` to see how generic structs were handled.
I made the following small change to an existing test program:
```
abas@ubuntu:~/p4lang/p4c$ git diff testdata/p4_16_samples/v1model-p4runtime-most-types1.p4
diff --git a/testdata/p4_16_samples/v1model-p4runtime-most-types1.p4 b/testdata/p4_16_samples/v1model-p4runtime-most-types1.p4
index 0399ebc47..ea45b57bc 100644
--- a/testdata/p4_16_samples/v1model-p4runtime-most-types1.p4
+++ b/testdata/p4_16_samples/v1model-p4runtime-most-types1.p4
@@ -173,6 +173,11 @@ struct struct1_t {
 #endif  // PROBLEM_NESTED_STRUCT_FIELD_TYPE
 }
 
+// generic stuct
+struct struct2_t<T> {
+       T data;
+}
+
 header custom_t {
     Eth0_t      addr0;
     Eth1_t      addr1;
@@ -255,6 +260,8 @@ header packet_out_header_t {
 #ifdef PROBLEM_SERENUM_FIELD
     serenum0_t   s0;
 #endif  // PROBLEM_SERENUM_FIELD
+
+    struct2_t<bit<32> > my_generic_struct2;
 }
 
 struct headers_t {
```

When running the test with the following command:
```bash
P4TEST_REPLACE=1 ctest -R p4/testdata/p4_16_samples/v1model-p4runtime-most-types1.p4 -V
```

I get the following crash:
```
terminate called after throwing an instance of 'Util::CompilerBug'
  what():  In file: /home/abas/p4lang/p4c/frontends/p4/typeMap.cpp:126
Compiler Bug: Could not find type for <StructField>(423526) _addr0/4204
```

If I run the same `p4test` invocation without the P4Runtime flags, there is no error.

Here is the stacktrace:
```
Generating P4Runtime output for architecture v1model
terminate called after throwing an instance of 'Util::CompilerBug'
  what():  In file: /home/abas/p4lang/p4c/frontends/p4/typeMap.cpp:126
Compiler Bug: Could not find type for <StructField>(423526) _addr0/4204


Thread 1 "p4test" received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
51	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007ffff694d7f1 in __GI_abort () at abort.c:79
#2  0x00007ffff71c1957 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007ffff71c7ae6 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007ffff71c7b21 in std::terminate() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007ffff71c7d54 in __cxa_throw () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00005555582bd98c in P4::TypeMap::getType (this=0x7fffffffd080, element=0x555559cd19c0, notNull=true) at /home/abas/p4lang/p4c/frontends/p4/typeMap.cpp:126
#7  0x0000555557b66c85 in P4::ControlPlaneAPI::P4RuntimeAnalyzer::addControllerHeader (this=0x7fffffffcd30, type=0x555559894000) at /home/abas/p4lang/p4c/control-plane/p4RuntimeSerializer.cpp:1029
#8  0x0000555557b49c42 in P4::ControlPlaneAPI::P4RuntimeAnalyzer::<lambda(const IR::Type_Header*)>::operator()(const IR::Type_Header *) const (__closure=0x7fffffffcc18, type=0x555559894000)
    at /home/abas/p4lang/p4c/control-plane/p4RuntimeSerializer.cpp:1850
...
```

@mbudiu-vmw I can dig deeper, but if you know what's wrong off the top of your head, it will probably be much faster...

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
Signed-off-by: Mihai Budiu <mbudiu@vmware.com>
Fixes #3329

### CODE REVIEW

**Comment by mbudiu-vmw** [comment]
I will amend the test to check for generic structs too.

File: testdata/p4_16_samples/issue3329-bmv2.p4 [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 Cisco Systems, Inc.
```
Comment (antoninbas) [inline testdata/p4_16_samples/issue3329-bmv2.p4:2]: Do you want to change this?

**Review by antoninbas — APPROVED** [review]
You don't have to add a generic struct to the test if not needed, I realized that I was testing the wrong thing in https://github.com/p4lang/p4c/issues/3329 :) Headers are always flattened, while I want to see how P4Info generation handles generic structs for things like digests. I'll write a new test.

**Comment by antoninbas** [comment]
Thanks for the fix @mbudiu-vmw !

**Pull request merged**